kratos/containers/variables_list_data_value_container.h

```diff
@@ -96,13 +95,12 @@ class KRATOS_API(KRATOS_CORE) VariablesListDataValueContainer
         // Setting the current position at the begining of data
         mpCurrentPosition = mpData;
 
-        SizeType size = mpVariablesList->DataSize();
-        for(VariablesList::const_iterator i_variable = mpVariablesList->begin() ;
-                i_variable != mpVariablesList->end() ; i_variable++)
-        {
+        const SizeType size = mpVariablesList->DataSize();
+        for(auto i_variable = mpVariablesList->begin(); i_variable != mpVariablesList->end() ; i_variable++) {
```

Comment:
Shouldn't you get the `cbegin` and `cend` to keep the previous const iterator?

Reply:
In this PR I am just moving to one line, not actually changing code

Reply:
IMO this is a code change

using `auto` here I am not sure if it is still a `const_iterator`

In my experience the compiler will try to avoid `const` unless it is forced

Reply:
https://github.com/KratosMultiphysics/Kratos/blob/88f57b7a2fbe135cc6599b7fd001fe907b8c5c83/kratos/containers/variables_list.h#L173

https://github.com/KratosMultiphysics/Kratos/blob/88f57b7a2fbe135cc6599b7fd001fe907b8c5c83/kratos/containers/variables_list.h#L177

begin and end are already const (there is not "not const" version )

Reply:
But if you use the auto with a non-const container you'll get a non-const iterator. If mpVariablesList is not const, which I assume is not in this case, you are changing the current behavior as @philbucher points.

Reply:
OK. I didn't see the there is "non const". I also agree that the auto doesn't change behavior in this case. Anyway I tend to agree @philbucher in this case. We either keep the current implementation or we use `const auto`.

Reply:
I am not sure if `const auto` works with iterators

I think this makes the iterator (which you want to increment/iterate) `const`  which gives a compiler error as opposed to `const_iterator` which is an iterator to `const`
As I said I am not 100% sure, please check to confirm if you want to know for sure

Reply:
No, you can't do const auto and a loop for

Reply:
http://cpp.sh/4rh5z This does not work

http://cpp.sh/5jyhp This works

Reply:
Ping @rubenzorrilla @philbucher

Reply:
as I said I am not in favor of this change, but I am not blocking

Reply:
I agree @philbucher. I'd keep the old implementation then.

Reply:
Done